dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -75,25 +76,35 @@ class DataManager(
      * @param storableDataSet contains all the inputs needed by Dataland
      * @return ID of the newly stored data in the data store
      */
-    @Transactional
     fun addDataSetToTemporaryStorageAndSendMessage(storableDataSet: StorableDataSet, correlationId: String):
         String {
+        val dataId = generateRandomDataId()
+        addDatasetToDatabase(dataId, storableDataSet, correlationId)
+        storeDataSetInTemporaryStoreAndSendMessage(dataId, storableDataSet, correlationId)
+        return dataId
+    }
+
+    /**
+     * Persists the data meta-information to the database ensuring that the database transaction
+     * ends directly after this function returns so that a MQ-Message might be sent out after this function completes
+     * @param dataId The dataId of the dataset to store
+     * @param storableDataSet the dataset to store
+     * @param correlationId the correlation id of the insertion process
+     */
+    @Transactional(propagation = Propagation.NEVER)
```

Comment:
Is my understanding correct, that setting Propagation.Never will prevent that a service can try to read a dataset will another service is still writing it? And by this resolving the issue we observed with the backend e2e tests.

Reply:
After a call i understood it. Conversation resolved